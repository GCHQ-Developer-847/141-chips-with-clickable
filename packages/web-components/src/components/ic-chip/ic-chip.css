@import "../../global/normalise.css";

:host {
  display: inline-block;
}

.chip {
  padding: var(--ic-space-xxs);
  font-size: 14px;
  border-radius: 80px;
  text-align: center;
  text-decoration: none;
}

.label {
  padding: 0 var(--ic-space-xs);
}

.container {
  display: inline-block;
}

.chip.dismissible {
  padding-right: calc(var(--ic-space-lg) + var(--ic-space-xxs));
  margin-right: calc((var(--ic-space-lg) + var(--ic-space-xxxs)) * -1);
}

.chip.dismissible.small {
  padding-right: calc(var(--ic-space-lg) + var(--ic-space-xxxs));
  margin-right: calc(var(--ic-space-lg) * -1);
}

.chip.dismissible.large {
  padding-right: var(--ic-space-xl);
  margin-right: calc((var(--ic-space-xl) - var(--ic-space-xxxs)) * -1);
}

.chip.small {
  padding: var(--ic-space-xxxs);
}

.chip.large {
  padding: var(--ic-space-xs);
}

.chip.dismissible .content-container {
  justify-content: space-between;
}

.content-container {
  height: 100%;
  justify-content: space-evenly;
}

.content-container,
.container {
  display: flex;
  align-items: center;
}

.chip.clickable {
  transition: var(--ic-easing-transition-fast);
}

.chip.clickable:focus,
.chip.clickable.focussed {
  box-shadow: var(--ic-border-focus);
  outline: var(--ic-hc-focus-outline);
  z-index: 1;
}

.chip.clickable:focus ~ ic-tooltip,
.chip.clickable.focussed ~ ic-tooltip {
  z-index: 1;
}

.filled {
  background-color: var(--ic-architectural-200);
  color: black;
  border: none;
}

.filled.disabled {
  background-color: var(--ic-architectural-100);
}

.filled.clickable:hover:not(:focus, .focussed),
.dismiss-icon:hover:not(:focus, .focussed) {
  background-color: var(--ic-architectural-300);
  cursor: pointer;
}

.filled.clickable:active:not(:focus, .focussed),
.dismiss-icon:active:not(:focus, .focussed) {
  background-color: var(--ic-architectural-400);
}

.outline {
  color: var(--ic-architectural-900);
  border: 1px solid var(--ic-architectural-900);
  padding: calc(var(--ic-space-xxs) - 1px);
}

.outline.small {
  padding: calc(var(--ic-space-xxxs) - 1px);
}

.outline.large {
  padding: calc(var(--ic-space-xs) - 1px);
}

.outline.clickable {
  background: none;
}

.outline.clickable:hover:not(:focus, .focussed),
.outline ~ ic-tooltip .dismiss-icon:hover:not(:focus, .focussed) {
  background-color: var(--ic-action-dark-bg-hover);
  cursor: pointer;
}

.outline.clickable:active:not(:focus, .focussed),
.outline ~ ic-tooltip .dismiss-icon:active:not(:focus, .focussed) {
  background-color: var(--ic-action-dark-bg-active);
}

.outline.disabled {
  border: var(--ic-border-disabled);
  background: none;
}

.chip.disabled ic-typography {
  color: var(--ic-architectural-200);
}

.dismiss-icon {
  border: none;
  border-radius: 50%;
  padding: 0;
  background: none;
  cursor: pointer;
  transition: var(--ic-easing-transition-fast);
}

.dismiss-icon,
ic-tooltip {
  height: 20px;
}

ic-tooltip {
  margin-right: calc(var(--ic-space-xxs) + var(--ic-space-xxxs));
}

.chip.small ~ ic-tooltip {
  margin-right: var(--ic-space-xxs);
}

.chip.large ~ ic-tooltip {
  margin-right: calc(var(--ic-space-xs) + var(--ic-space-xxxs));
}

.dismiss-icon:focus {
  box-shadow: var(--ic-border-focus);
  outline: var(--ic-hc-focus-outline);
}

.dismiss-icon[disabled] {
  pointer-events: none;
  color: var(--ic-architectural-200);
}

.icon {
  width: var(--ic-space-lg);
  height: var(--ic-space-lg);
  padding: var(--ic-space-xxxs);
  box-sizing: border-box;
}

.chip.dismissible.small .icon {
  padding-left: var(--ic-space-xxxs);
}

.chip.disabled path,
.chip.disabled ::slotted(svg) {
  fill: var(--ic-architectural-200);
}

@media (forced-colors: active) {
  .chip {
    border: var(--ic-hc-border);
  }
}
