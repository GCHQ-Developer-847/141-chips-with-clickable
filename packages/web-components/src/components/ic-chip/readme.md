# ic-chip



<!-- Auto Generated Below -->


## Properties

| Property             | Attribute        | Description                                                                                     | Type                                                                                                                                                                                     | Default     |
| -------------------- | ---------------- | ----------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `appearance`         | `appearance`     | The emphasis of the chip.                                                                       | `"filled" \| "outline"`                                                                                                                                                                  | `"filled"`  |
| `clickable`          | `clickable`      | If `true`, the chip will be clickable instead of static.                                        | `boolean`                                                                                                                                                                                | `false`     |
| `disabled`           | `disabled`       | If `true`, and the chip is clickable, it will be disabled.                                      | `boolean`                                                                                                                                                                                | `false`     |
| `dismissible`        | `dismissible`    | If `true`, the chip will have a close button at the end to dismiss it.                          | `boolean`                                                                                                                                                                                | `false`     |
| `href`               | `href`           | The URL for clickable chips.                                                                    | `string`                                                                                                                                                                                 | `undefined` |
| `hreflang`           | `hreflang`       | The human-readable explanation of the URL.                                                      | `string`                                                                                                                                                                                 | `""`        |
| `label` _(required)_ | `label`          | The text rendered within the chip.                                                              | `string`                                                                                                                                                                                 | `undefined` |
| `referrerpolicy`     | `referrerpolicy` | How much of the referrer to send when following the link.                                       | `"" \| "no-referrer" \| "no-referrer-when-downgrade" \| "origin" \| "origin-when-cross-origin" \| "same-origin" \| "strict-origin" \| "strict-origin-when-cross-origin" \| "unsafe-url"` | `undefined` |
| `rel`                | `rel`            | The relationship of the linked URL as space-separated link types.                               | `string`                                                                                                                                                                                 | `undefined` |
| `size`               | `size`           | The size of the chip.                                                                           | `"default" \| "large" \| "small"`                                                                                                                                                        | `"default"` |
| `target`             | `target`         | Where to display the linked URL, as the name for a browsing context (a tab, window, or iframe). | `string`                                                                                                                                                                                 | `undefined` |


## Events

| Event     | Description                                  | Type                |
| --------- | -------------------------------------------- | ------------------- |
| `dismiss` | Is emitted when the user dismisses the chip. | `CustomEvent<void>` |


## Slots

| Slot     | Description                                        |
| -------- | -------------------------------------------------- |
| `"icon"` | Content will be rendered at the start of the chip. |


## Dependencies

### Depends on

- [ic-typography](../ic-typography)
- [ic-tooltip](../ic-tooltip)

### Graph
```mermaid
graph TD;
  ic-chip --> ic-typography
  ic-chip --> ic-tooltip
  ic-tooltip --> ic-typography
  style ic-chip fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


