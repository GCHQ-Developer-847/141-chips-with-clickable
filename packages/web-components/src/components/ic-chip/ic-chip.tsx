import {
  Component,
  h,
  Prop,
  State,
  Listen,
  Event,
  EventEmitter,
  Element,
} from "@stencil/core";
import {
  onComponentRequiredPropUndefined,
  isSlotUsed,
} from "../../utils/helpers";
import { IcChipAppearance, IcChipSizes } from "./ic-chip.types";
import dismissIcon from "../../assets/dismiss-icon.svg";

/**
 * @slot icon - Content will be rendered at the start of the chip.
 */
@Component({
  tag: "ic-chip",
  styleUrl: "ic-chip.css",
  shadow: true,
})
export class Chip {
  @Element() el: HTMLIcChipElement;
  /**
   * If `true`, the chip will be clickable instead of static.
   */
  @Prop({ mutable: true }) clickable?: boolean = false;
  /**
   * The text rendered within the chip.
   */
  @Prop() label!: string;
  /**
   * The emphasis of the chip.
   */
  @Prop() appearance?: IcChipAppearance = "filled";
  /**
   * The size of the chip.
   */
  @Prop() size?: IcChipSizes = "default";
  /**
   * If `true`, the chip will have a close button at the end to dismiss it.
   */
  @Prop() dismissible?: boolean = false;
  /**
   * The URL for clickable chips.
   */
  @Prop() href?: string | undefined;
  /**
   * The human-readable explanation of the URL.
   */
  @Prop() hreflang?: string = "";
  /**
   * How much of the referrer to send when following the link.
   */
  @Prop() referrerpolicy?: ReferrerPolicy;
  /**
   * The relationship of the linked URL as space-separated link types.
   */
  @Prop() rel?: string;
  /**
   * Where to display the linked URL, as the name for a browsing context (a tab, window, or iframe).
   */
  @Prop() target?: string;
  /**
   * If `true`, and the chip is clickable, it will be disabled.
   */
  @Prop() disabled?: boolean = false;

  @State() parentIsAnchorTag: boolean = false;

  @State() parentEl: HTMLElement | null = null;

  @State() visible: boolean = true;

  @State() isFocussed: boolean = false;

  @Listen("click", { capture: true })
  handleHostClick(event: Event): void {
    if (this.disabled) {
      event.stopImmediatePropagation();
    }
  }

  @Listen("dismiss", { capture: true })
  handleClick(): void {
    this.visible = !this.visible;
  }
  /**
   * Is emitted when the user dismisses the chip.
   */
  @Event() dismiss: EventEmitter<void>;

  private dismissAction = (event: MouseEvent): void => {
    event.stopPropagation();
    this.dismiss.emit();
  };

  private parentFocussed = (): void => {
    this.isFocussed = true;
  };

  private parentBlurred = (): void => {
    this.isFocussed = false;
  };

  componentWillLoad(): void {
    this.parentEl = this.el.parentElement;

    if (this.parentEl.tagName === "A") {
      this.clickable = true;
      this.parentIsAnchorTag = true;
      this.parentEl.addEventListener("focus", this.parentFocussed);
      this.parentEl.addEventListener("blur", this.parentBlurred);
    }
  }

  disconnectedCallback(): void {
    if (this.parentIsAnchorTag) {
      this.parentEl.removeEventListener("focus", this.parentFocussed);
      this.parentEl.removeEventListener("blur", this.parentBlurred);
    }
  }

  componentDidLoad(): void {
    onComponentRequiredPropUndefined(
      [{ prop: this.label, propName: "label" }],
      "Chip"
    );
  }

  render() {
    const {
      label,
      appearance,
      size,
      dismissible,
      visible,
      clickable,
      disabled,
      parentIsAnchorTag,
      isFocussed,
      href,
      hreflang,
      referrerpolicy,
      rel,
      target,
    } = this;

    const Component = parentIsAnchorTag
      ? "div"
      : clickable
      ? this.href === undefined
        ? "button"
        : "a"
      : "div";

    const attrs = Component == "a" && {
      href: href,
      hrefLang: hreflang,
      referrerPolicy: referrerpolicy,
      rel: rel,
      target: target,
    };

    return (
      visible && (
        <div class="container">
          <Component
            class={{
              ["chip"]: true,
              [`${appearance}`]: true,
              [`${size}`]: true,
              ["clickable"]: clickable && !disabled,
              ["disabled"]: disabled,
              ["focussed"]: isFocussed,
              ["dismissible"]: dismissible,
            }}
            tabindex={clickable && !parentIsAnchorTag ? 0 : -1}
            aria-disabled={disabled ? "true" : null}
            disabled={disabled ? true : null}
            {...attrs}
          >
            <div class="content-container">
              {isSlotUsed(this.el, "icon") && (
                <div class="icon">
                  <slot name="icon" />
                </div>
              )}
              <ic-typography
                variant="label"
                apply-vertical-margins={false}
                class="label"
              >
                <span>{label}</span>
              </ic-typography>
            </div>
          </Component>
          {dismissible && (
            <ic-tooltip
              label="Dismiss"
              target="dismiss-icon"
              class={{ ["tooltip-disabled"]: disabled }}
            >
              <button
                id="dismiss-icon"
                class={{
                  ["dismiss-icon"]: true,
                }}
                aria-label={`Dismiss ${label}`}
                disabled={disabled}
                tabindex={disabled ? -1 : 0}
                onClick={this.dismissAction}
                innerHTML={dismissIcon}
              ></button>
            </ic-tooltip>
          )}
        </div>
      )
    );
  }
}
