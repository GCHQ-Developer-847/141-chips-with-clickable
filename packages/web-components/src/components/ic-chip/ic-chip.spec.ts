import { newSpecPage } from "@stencil/core/testing";
import { Chip } from "./ic-chip";

describe("ic-chip component renders label", () => {
  it("should render static", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<ic-chip label="Label">
            <svg slot="icon" width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"/>
</svg>
            </ic-chip>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render dismissible", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<ic-chip label="Label" dismissible="true">
            <svg slot="icon" width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"/>
</svg>
            </ic-chip>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render disabled", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<ic-chip label="Label" dismissible="true" clickable="true" disabled="true">
            <svg slot="icon" width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"/>
</svg>
            </ic-chip>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render small", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<ic-chip label="Label" dismissible="true" clickable="true" size="small">
              <svg slot="icon" width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"/>
  </svg>
              </ic-chip>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render clickable", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<ic-chip label="Label" clickable="true" dismissible="true" href="/">
            <svg slot="icon" width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"/>
</svg>
            </ic-chip>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a link parent", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<a href="/"><ic-chip label="Label" dismissible="true" clickable=true></ic-chip></a>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should apply 'focussed' style when parent is focussed", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<a href="/"><ic-chip id="test-id" label="Label" dismissible="true" clickable=true></ic-chip></a>`,
    });

    expect(page.root).not.toBeNull;

    const element = await document.getElementById("test-id");

    await expect(element.classList.contains("focussed")).toBeFalsy;

    await element.focus();

    await expect(element.classList.contains("focussed")).toBeTruthy;
  });

  it("should lose 'focussed' style when parent loses focus", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<a href="/"><ic-chip id="test-chip" label="Label" dismissible="true" clickable=true></ic-chip></a>`,
    });

    expect(page.root).not.toBeNull;

    const element = await document.getElementById("test-chip");

    await element.focus();

    await expect(element.classList.contains("focussed")).toBeTruthy;

    await element.blur();

    await expect(element.classList.contains("focussed")).toBeFalsy;

    await page.rootInstance.disconnectedCallback();
  });

  it("should stop immediate propagation of a click event when disabled", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<ic-chip id="test-chip" label="Label" dismissible="true" clickable=true disabled=true onclick="alert('test')"></ic-chip>`,
    });

    jest.spyOn(window, "alert").mockImplementation();

    const element = await document.getElementById("test-chip");

    element.click();

    await page.waitForChanges();

    await expect(window.alert).not.toHaveBeenCalled;
  });

  it("should close on dismiss icon click", async () => {
    const page = await newSpecPage({
      components: [Chip],
      html: `<ic-chip label="This is dismissible" dismissible="true"></ic-chip>`,
    });

    let chip = await page.root.shadowRoot.querySelector("div.container");
    const dismissButton = await page.root.shadowRoot.querySelector("button");

    expect(chip).not.toBeNull();
    expect(dismissButton).not.toBeNull();

    await dismissButton.click();

    await page.waitForChanges();
    chip = await page.root.shadowRoot.querySelector("div.container");

    expect(chip).toBeNull();
  });
});
