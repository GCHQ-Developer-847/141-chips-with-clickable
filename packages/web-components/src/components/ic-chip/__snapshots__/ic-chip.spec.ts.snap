// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-chip component renders label should render clickable 1`] = `
<ic-chip clickable="true" dismissible="true" href="/" label="Label">
  <mock:shadow-root>
    <div class="container">
      <a class="chip clickable default dismissible filled" href="/" hreflang="" tabindex="0">
        <div class="content-container">
          <div class="icon">
            <slot name="icon"></slot>
          </div>
          <ic-typography class="label" variant="label">
            <span>
              Label
            </span>
          </ic-typography>
        </div>
      </a>
      <ic-tooltip label="Dismiss" target="dismiss-icon">
        <button aria-label="Dismiss Label" class="dismiss-icon" id="dismiss-icon" tabindex="0">
          svg
        </button>
      </ic-tooltip>
    </div>
  </mock:shadow-root>
  <svg fill="none" height="20" slot="icon" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
    <path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"></path>
  </svg>
</ic-chip>
`;

exports[`ic-chip component renders label should render disabled 1`] = `
<ic-chip clickable="true" disabled="true" dismissible="true" label="Label">
  <mock:shadow-root>
    <div class="container">
      <button aria-disabled="true" class="chip default disabled dismissible filled" disabled="" tabindex="0">
        <div class="content-container">
          <div class="icon">
            <slot name="icon"></slot>
          </div>
          <ic-typography class="label" variant="label">
            <span>
              Label
            </span>
          </ic-typography>
        </div>
      </button>
      <ic-tooltip class="tooltip-disabled" label="Dismiss" target="dismiss-icon">
        <button aria-label="Dismiss Label" class="dismiss-icon" disabled="" id="dismiss-icon" tabindex="-1">
          svg
        </button>
      </ic-tooltip>
    </div>
  </mock:shadow-root>
  <svg fill="none" height="20" slot="icon" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
    <path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"></path>
  </svg>
</ic-chip>
`;

exports[`ic-chip component renders label should render dismissible 1`] = `
<ic-chip dismissible="true" label="Label">
  <mock:shadow-root>
    <div class="container">
      <div class="chip default dismissible filled" tabindex="-1">
        <div class="content-container">
          <div class="icon">
            <slot name="icon"></slot>
          </div>
          <ic-typography class="label" variant="label">
            <span>
              Label
            </span>
          </ic-typography>
        </div>
      </div>
      <ic-tooltip label="Dismiss" target="dismiss-icon">
        <button aria-label="Dismiss Label" class="dismiss-icon" id="dismiss-icon" tabindex="0">
          svg
        </button>
      </ic-tooltip>
    </div>
  </mock:shadow-root>
  <svg fill="none" height="20" slot="icon" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
    <path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"></path>
  </svg>
</ic-chip>
`;

exports[`ic-chip component renders label should render small 1`] = `
<ic-chip clickable="true" dismissible="true" label="Label" size="small">
  <mock:shadow-root>
    <div class="container">
      <button class="chip clickable dismissible filled small" tabindex="0">
        <div class="content-container">
          <div class="icon">
            <slot name="icon"></slot>
          </div>
          <ic-typography class="label" variant="label">
            <span>
              Label
            </span>
          </ic-typography>
        </div>
      </button>
      <ic-tooltip label="Dismiss" target="dismiss-icon">
        <button aria-label="Dismiss Label" class="dismiss-icon" id="dismiss-icon" tabindex="0">
          svg
        </button>
      </ic-tooltip>
    </div>
  </mock:shadow-root>
  <svg fill="none" height="20" slot="icon" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
    <path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"></path>
  </svg>
</ic-chip>
`;

exports[`ic-chip component renders label should render static 1`] = `
<ic-chip label="Label">
  <mock:shadow-root>
    <div class="container">
      <div class="chip default filled" tabindex="-1">
        <div class="content-container">
          <div class="icon">
            <slot name="icon"></slot>
          </div>
          <ic-typography class="label" variant="label">
            <span>
              Label
            </span>
          </ic-typography>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <svg fill="none" height="20" slot="icon" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg">
    <path d="M10 0C4.48 0 0 4.48 0 10C0 15.52 4.48 20 10 20C15.52 20 20 15.52 20 10C20 4.48 15.52 0 10 0ZM10 3C11.66 3 13 4.34 13 6C13 7.66 11.66 9 10 9C8.34 9 7 7.66 7 6C7 4.34 8.34 3 10 3ZM10 17.2C7.5 17.2 5.29 15.92 4 13.98C4.03 11.99 8 10.9 10 10.9C11.99 10.9 15.97 11.99 16 13.98C14.71 15.92 12.5 17.2 10 17.2Z" fill="currentColor"></path>
  </svg>
</ic-chip>
`;

exports[`ic-chip component renders label should render with a link parent 1`] = `
<ic-chip clickable="true" dismissible="true" label="Label">
  <mock:shadow-root>
    <div class="container">
      <div class="chip clickable default dismissible filled" tabindex="-1">
        <div class="content-container">
          <ic-typography class="label" variant="label">
            <span>
              Label
            </span>
          </ic-typography>
        </div>
      </div>
      <ic-tooltip label="Dismiss" target="dismiss-icon">
        <button aria-label="Dismiss Label" class="dismiss-icon" id="dismiss-icon" tabindex="0">
          svg
        </button>
      </ic-tooltip>
    </div>
  </mock:shadow-root>
</ic-chip>
`;
